pkg/dfns-api-client/auth_transport.go

```diff
@@ -0,0 +1,276 @@
+package dfns_api_client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"slices"
+	"strconv"
+
+	"github/dfns/dfns-sdk-go/pkg/credentials"
+)
+
+const (
+	UserActionHeader    = "x-dfns-useraction"
+	appIDHeader         = "x-dfns-appid"
+	nonceHeader         = "x-dfns-nonce"
+	authorizationHeader = "authorization"
+	contentTypeHeader   = "Content-Type"
+)
+
+type authTransport struct {
+	*DfnsApiOptions
+}
+
+func (auth *authTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	auth.setHeaders(req)
+
+	if ok := slices.Contains([]string{"POST", "PUT", "DELETE"}, req.Method); ok {
+		performUserAction, err := shouldPerformUserAction(req)
+		if err != nil {
+			return nil, err
+		}
+
+		if performUserAction {
+			if err := auth.performUserActionRequest(req); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	return executeRequest(req)
+}
+
+func shouldPerformUserAction(req *http.Request) (bool, error) {
+	authHeaderStr := req.Header.Get(UserActionHeader)
+	if authHeaderStr == "" {
+		return true, nil
+	}
+	defer req.Header.Del(UserActionHeader)
+
```

Comment:
Can `authHeaderStr` be `nil` as well ?

Reply:
Unfortunately no :/ `req.Header.Get(UserActionHeader)` will always return a string (an empty one if not set)

Reply:
The question was more to verify if "there is a possibility that it will be nil", not really "can you make it null here" :)